prompting/apparmor/message.go

```diff
@@ -0,0 +1,436 @@
+package apparmor
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/snapcore/snapd/arch"
+)
+
+// Message fields are defined as raw sized integer types as the same type may be
+// packed as 16 bit or 32 bit integer, to accommodate other fields in the
+// structure.
+
+// MsgHeader is the header of all apparmor notification messages.
+//
+// The header encodes the size of the entire message, including variable-size
+// components and the version of the notification protocol.
+type MsgHeader struct {
+	// Length is the length of the entire message, including any more
+	// specialized messages appended after the header and any variable-length
+	// objects referenced by any such message.
+	Length uint16
+	// Version is the version of the communication protocol.
+	// Currently version 2 is implemented by the demo kernel based on Linux 5.7.7,
+	// but version 3 was used in the past.
+	Version uint16
+}
+
+// UnmarshalBinary unmarshals the header from binary form.
+func (msg *MsgHeader) UnmarshalBinary(data []byte) error {
+	const prefix = "cannot unmarshal apparmor message header"
+
+	// Unpack fixed-size elements.
+	order := arch.Endian() // ioctl messages are native byte order, but others might not be
```

Comment:
(nitpick) what are the `others` that `but others might not be` refers to? Maybe either spell it out of just leave this part out?

Reply:
What I meant by `others` was other non-ioctl messages. In case the message.go code gets used for some other syscall, I suppose. I changed the message: `s/but others might not be/verify endianness if using for other messages/g`. Happy to revise or remove the comment if it is confusing, or if this will only ever be used for ioctl messages.